packages/labs/context/src/lib/decorators/context-provider.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license
+ * Copyright 2017 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {ReactiveElement} from '@lit/reactive-element';
+import {decorateProperty} from '@lit/reactive-element/decorators/base.js';
+import {ContextKey} from '../context-key.js';
+import {ContextProvider} from '../controllers/context-provider.js';
+
+/*
+ * IMPORTANT: For compatibility with tsickle and the Closure JS compiler, all
+ * property decorators (but not class decorators) in this file that have
+ * an @ExportDecoratedItems annotation must be defined as a regular function,
+ * not an arrow function.
+ */
+
+/**
+ * A property decorator that adds a ContextConsumer controller to the component
+ * which will try and retrieve a value for the property via the Context API.
+ *
+ * @param context A Context identifier value created via `createContext`
+ * @param multiple An optional boolean which when true allows the value to be updated
+ *   multiple times.
+ *
+ * See: https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector
+ *
+ * @example
+ *
+ * ```ts
+ * import {loggerContext} from 'community-protocols/logger';
+ *
+ * class MyElement {
+ *   @contextProvided(loggerContext)
+ *   logger;
+ *
+ *   doThing() {
+ *     this.logger.log('thing was done');
+ *   }
```

Comment:
The specification says that ContextAPI *should not* consider *Dependency Injection*.  This looks like it.

Maybe we should change the examples to something that isn't part of what the specification. Say a component with two properties

```ts

export interface IUserData {
  readonly id: string
}

// Say another slice of state is about status
export interface IUserStatus extends IUserData {
  readonly presence: 'offline' | 'available' | 'away'
}

export interface IUserInfo extends IUserData {
  readonly firstName: string
}

export const UserInfoContext = createContext<IUserInfo>('user-info')

class UserInfoNameFormatter extends LitElement {
  @property()
  id = ''

  @state()
  userInfo

  public constructor() {
    super()
    new ContextConsumer(this, UserInfoContext, (value: IUserInfo) => {
      if (value.id === this.id) {
        this.userInfo = { ...value }
      }
    })
  }

  render() {
    return html`<div data-id=${this.id}>${this.firstName}</div>`
  }
}
 ```

Reply:
In the interests of merging I'm going to defer this, filed https://github.com/lit/lit/issues/2764 to track